pkg/network/event_windows.go

```diff
@@ -45,6 +49,118 @@ const (
 	UDPProtocol = 17
 )
 
+var (
+	ephemeralRanges = map[ConnectionFamily]map[ConnectionType]map[string]uint16{
+		AFINET: {
+			UDP: {
+				"lo": 0,
+				"hi": 0,
+			},
+			TCP: {
+				"lo": 0,
+				"hi": 0,
+			},
+		},
+		AFINET6: {
+			UDP: {
+				"lo": 0,
+				"hi": 0,
+			},
+			TCP: {
+				"lo": 0,
+				"hi": 0,
+			},
+		},
+	}
+)
+
+func init() {
+	var families = [...]ConnectionFamily{AFINET, AFINET6}
+	var protos = [...]ConnectionType{UDP, TCP}
+	for _, f := range families {
+		for _, p := range protos {
+			l, h, err := getEphemeralRange(f, p)
+			if err == nil {
+				ephemeralRanges[f][p]["lo"] = l
+				ephemeralRanges[f][p]["hi"] = h
+			}
+		}
+	}
+}
+func getEphemeralRange(f ConnectionFamily, t ConnectionType) (low, hi uint16, err error) {
+	var protoarg string
+	var familyarg string
+	switch f {
+	case AFINET6:
+		familyarg = "ipv6"
+	default:
+		familyarg = "ipv4"
+	}
+	switch t {
+	case TCP:
+		protoarg = "tcp"
+	default:
+		protoarg = "udp"
+	}
+	cmd := exec.Command("netsh", "int", familyarg, "show", "dynamicport", protoarg)
+	cmdOutput := &bytes.Buffer{}
+	// output should be of the format
+	/*
+		Protocol tcp Dynamic Port Range
+		---------------------------------
+		Start Port      : 49000
+		Number of Ports : 16000
+	*/
+	cmd.Stdout = cmdOutput
+	err = cmd.Run()
+	if err != nil {
+		return
+	}
+	output := cmdOutput.Bytes()
+	var startPortLine = regexp.MustCompile(`Start.*: (\d+)`)
+	var numberLine = regexp.MustCompile(`Number.*: (\d+)`)
+
+	startPort := startPortLine.FindStringSubmatch(string(output))
+	rangeLen := numberLine.FindStringSubmatch(string(output))
+
+	portstart, err := strconv.Atoi(startPort[1])
+	if err != nil {
+		return
+	}
```

Comment:
@stephengroat-dd interesting that this did not get picked up by the CodeQL run.  Perhaps because it is +windows?